PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -1226,6 +1226,24 @@ void ScintillaEditView::setLexer(int lexerID, LangType langType, int whichList)
 		const char * keyWords_char = wmc.wchar2char(pKwArray[LANG_INDEX_TYPE5], CP_ACP);
 		setKeywords(langType, keyWords_char, LANG_INDEX_TYPE5);
 	}
+
+	if (whichList & LIST_7)
+	{
+		const char * keyWords_char = wmc.wchar2char(pKwArray[LANG_INDEX_TYPE6], CP_ACP);
+		setKeywords(langType, keyWords_char, LANG_INDEX_TYPE6);
+	}
+
+	if (whichList & LIST_8)
+	{
+		const char * keyWords_char = wmc.wchar2char(pKwArray[LANG_INDEX_TYPE7], CP_ACP);
+		setKeywords(langType, keyWords_char, LANG_INDEX_TYPE7);
+	}
+
+	if (whichList & LIST_9)

```

Comment:
`You can set up to 9 lists of keywords for use by the current lexer. keyWordSet can be 0 to 8:` 
https://www.scintilla.org/ScintillaDoc.html#SCI_SETKEYWORDS

Reply:
"_keyWordSet can be 0 to 8 (actually 0 to **KEYWORDSET_MAX**)_"

In Scintilla.h: `#define KEYWORDSET_MAX 30`.

Should be OK, depending on lexer (BaanC?) ?

Reply:
@donho I have removed LIST_9.
However, I would like to highlight that [KEYWORDSET_MAX](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/92bad0a60ad606b30df9ed97aecf4ff27bb6e967/scintilla/include/Scintilla.h#L1003) is currently 30... but, BAANC is using the Maximum Keyword sets, which will be fulfilled with LIST_8.